### ISSUE TITLE
(Need Help) Encountering Warnings When Using Nuxt3 + shadcnUI and Unsure How to Fix Them

### ISSUE THREAD

**Issue by one-222333** [nuxt/nuxt#29002]
**(Need Help) Encountering Warnings When Using Nuxt3 + shadcnUI and Unsure How to Fix Them**

Here’s the component code:

```vue
<script setup lang="ts">
import { DialogRoot, VisuallyHidden } from "radix-vue";
</script>

<template>
  <div id="index-side-bar">
    <slot />
    <Sheet :default-open="true">
      <SheetContent :side="'left'" class="outline-0 p-0 w-[240px] z-[99999]">
        <VisuallyHidden>
          <SheetHeader>
            <SheetTitle>dialog</SheetTitle>
            <SheetDescription>description</SheetDescription>
          </SheetHeader>
        </VisuallyHidden>
        <div style="width: 100%; height: 100vh; display: flex; flex-direction: column;">
          <!-- Header -->
          <div class="w-full h-[56px] flex items-center sm:h-[48px]">
            <div class="w-full h-[56px] flex items-center ml-[14px] sm:ml-[16px]">
              <div class="w-10 h-10 flex items-center justify-center sm:hidden">
                <!-- SVG Icon Placeholder -->
              </div>
              <div class="w-[120px] h-[56px] sm:w-auto sm:h-[48px]" style="display: flex; justify-content: center; align-items: center;">
                <div class="w-[94px] h-[30px]">
                  <!-- Logo Placeholder -->
                </div>
              </div>
            </div>
          </div>
          <!-- Footer -->
          <div class="sidebar_toolbar">
            <SheetClose class="outline-0">
              <NuxtLayout name="tab-toolbar-nav-menu" />
            </SheetClose>
          </div>
        </div>
      </SheetContent>
    </Sheet>
  </div>
</template>
```

ShadcnUI   Sheet.vue:

```vue
<script setup lang="ts">
import { DialogRoot, type DialogRootEmits, type DialogRootProps, useForwardPropsEmits } from 'radix-vue';

const props = defineProps<DialogRootProps>();
const emits = defineEmits<DialogRootEmits>();
const forwarded = useForwardPropsEmits(props, emits);
</script>

<template>
  <DialogRoot v-bind="forwarded">
    <slot />
  </DialogRoot>
</template>
```

**The warning message:**

```text
[Vue warn]: Extraneous non-props attributes (data-v-e01f9c95) were passed to component but could not be automatically inherited because component renders fragment or text root nodes.
  at <DialogRoot defaultOpen=true onUpdate:open=fn data-v-e01f9c95="" >
  at <Sheet default-open=true >
  at <SideBar ref=Ref< undefined > >
```

This warning has appeared multiple times and only occurs on the server side. No warnings appear on the client side.

Through testing, I found that switching to the `DialogRoot` component eliminates the warnings. The warnings reappear when using the `Sheet` component.

Here’s the alternative component:

```vue
<script setup lang="ts">
import { DialogRoot, VisuallyHidden } from "radix-vue";
</script>

<template>
  <div id="index-side-bar">
    <slot />
    <DialogRoot :default-open="true">
      <SheetContent :side="'left'" class="outline-0 p-0 w-[240px] z-[99999]">
        <VisuallyHidden>
          <SheetHeader>
            <SheetTitle>dialog</SheetTitle>
            <SheetDescription>description</SheetDescription>
          </SheetHeader>
        </VisuallyHidden>
        <div style="width: 100%; height: 100vh; display: flex; flex-direction: column;">
          <!-- Header -->
          <div class="w-full h-[56px] flex items-center sm:h-[48px]">
            <div class="w-full h-[56px] flex items-center ml-[14px] sm:ml-[16px]">
              <div class="w-10 h-10 flex items-center justify-center sm:hidden">
                <!-- SVG Icon Placeholder -->
              </div>
              <div class="w-[120px] h-[56px] sm:w-auto sm:h-[48px]" style="display: flex; justify-content: center; align-items: center;">
                <div class="w-[94px] h-[30px]">
                  <!-- Logo Placeholder -->
                </div>
              </div>
            </div>
          </div>
          <!-- Footer -->
          <div class="sidebar_toolbar">
            <SheetClose class="outline-0">
              <NuxtLayout name="tab-toolbar-nav-menu" />
            </SheetClose>
          </div>
        </div>
      </SheetContent>
    </DialogRoot>
  </div>
</template>
```

**When using DevTools**, the same warning occurs, similar to the server-side warning, but it doesn’t appear on the client side:

```text
[Vue warn]: Extraneous non-props attributes (data-v-inspector) were passed to component but could not be automatically inherited because component renders fragment or text root nodes.
  at <DialogRoot default-open=true data-v-inspector="layouts/index/SideBar.vue:16:7" >
  at <SideBar data-v-inspector="app.vue:4:7" ref=Ref< undefined > >
  at <LayoutLoader key="index-side-bar" layoutProps= { 'data-v-inspector': 'app.vue:4:7',
```

I have already disabled DevTools with `componentInspector: false`.

GitHub link: [Nuxt DevTools Issue #722](https://github.com/nuxt/devtools/issues/722).

**Comment by danielroe** [comment]
This should probably be raised with `radix-vue` - it's a consequence of Vue 3.5 and is not related to Nuxt.

**Comment by mrleblanc101** [comment]
I have this issue and I don't have Radix-vue. In fact I don't have any UI libraries:
```
    "devDependencies": {
        "@nuxt/image": "^1.2.0",
        "@nuxtjs/google-fonts": "^3.1.3",
        "nuxt": "^3.13.0",
        "sass": "^1.70.0",
        "vue": "^3.3.4",
        "vue-router": "^4.2.5"
    },
    "dependencies": {
        "@nuxt/content": "^2.10.0",
        "body-scroll-lock": "vercel/body-scroll-lock",
        "gsap": "^3.12.5",
        "normalize.css": "^8.0.1"
    }
```

>  WARN  [Vue warn]: Extraneous non-props attributes (data-v-inspector) were passed to component but could not be automatically inherited because component renders fragment or text root nodes. 
  at <DevOnly data-v-inspector="app.vue:2:5" > 
  at <App data-v-inspector="node_modules/nuxt/dist/app/components/nuxt-root.vue:16:5" > 
  at <NuxtRoot>

**Comment by danielroe** [comment]
@mrleblanc101 That has been resolved in https://github.com/nuxt/nuxt/pull/28939 and should be in the next Nuxt release.

**Comment by JorgeO3** [comment]
I have the same error and I just updated to nuxt version 3.13.2, will it be fixed in version 3.13.3?

**Comment by JorgeO3** [comment]
I have the same error and I just updated to nuxt version 3.13.2, will it be fixed in version 3.13.3?

**Comment by danielroe** [comment]
It needs to be fixed in radix-vue.
